core/unix/os.c

```diff
@@ -7245,16 +7245,18 @@ pre_system_call(dcontext_t *dcontext)
         } data_t;
         dcontext->sys_param3 = sys_param(dcontext, 5);
         data_t *data_param = (data_t *)dcontext->sys_param3;
-        data_t data;
+        data_t data = { 0 };
         /* Refer to comments in SYS_ppoll above. Taking extra steps here due to struct
          * argument in pselect6.
          */
-        if (!d_r_safe_read(data_param, sizeof(data), &data)) {
-            LOG(THREAD, LOG_SYSCALLS, 2, "\treturning EFAULT to app for pselect6\n");
-            set_failure_return_val(dcontext, EFAULT);
-            DODEBUG({ dcontext->expect_last_syscall_to_fail = true; });
-            execute_syscall = false;
-            break;
+        if (data_param != NULL) {
```

Comment:
Hmm, I would have done what seems simpler and easier to read logic to me:
```C
if (data_param == NULL) {
    /* The kernel does not consider a NULL 6th+7th-args struct to be an error but just a NULL sigmask. */
    dcontext->sys_param4 = NULL;
    break;
}
```
WDYT?  Then no lifetime worries for code below, no worries about zero being the right value for missing fields in data_t, etc.

Reply:
Ok, done.